spec/player_ai_minimax_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'player_ai_minimax'
+require 'game_board'
+require 'game_state'
+require 'game_rules'
+
+describe PlayerAiMinimax do
+  let (:io) { double }
+  let (:board) { GameBoard.new }
+  let (:rules) { GameRules.new }
+  let (:test_ai) { PlayerAiMinimax.new(:X, rules) }
+  let (:test_ai2) { PlayerAiMinimax.new(:O, rules) }
+  let (:test_state) { GameState.new(board , :X, :O) }
+  let(:spaces) { test_state.game_board.spaces }
+  let(:corners) { [0, 2, 6, 8] }
+
+  describe "It has a mark " do
+    it ".mark returns :X " do
+      expect(test_ai.mark).to eq(:X)
+    end
+  end
+  
+  describe "Playing the last move " do
+    it "plays 0 when that is the only space open " do
+      [1,2,3,4,5,6,7,8].each do |i|
+        spaces[i] = i
+      end
+      expect(test_ai.play_move(test_state)).to eq(0)
+    end
+    it "plays 1 when that is the only space open " do
+      [0,2,3,4,5,6,7,8].each do |i|
+        spaces[i] = i
+      end
+      expect(test_ai.play_move(test_state)).to eq(1)
+    end
+  end
+
+  describe "Playing second to last move " do
+    it "when only 7 and 8 are left it plays 7 when that will block opponent and 8 will not " do
+      [1,2,3,4].each do |i|
+        spaces[i] = :X
+      end
+      [0,5,6].each do |i|
```

Comment:
Not sure if you know this but gonna mention it anyway: in ruby you can do single-line blocks for simple things. i.e.
`[0,5,6].each { |i| spaces[i] = :O }`